src/postfork.cpp

```diff
@@ -112,7 +92,7 @@ bool child_set_group(job_t *j, process_t *p) {
             retval = false;
         }
     } else {
-        // This is probably stays unused in the child.
+        // The child does not actually use this field.
```

Comment:
I think we can drop the `else` branch completely since we're cleaning up the 2012 `getpgid` logic above? We just called assigned the value of `j->pgid` to own pgid, and that call didn't fail, so there's no need for the kernel context switch here again since the goal is to optimize.